model/modeldecoder/v2/decoder.go

```diff
@@ -72,7 +72,7 @@ var (
 	// reForServiceTargetExpr regex will capture service target type and name
 	// Service target type comprises of only lowercase alphabets
 	// Service target name comprises of all word characters
-	reForServiceTargetExpr = regexp.MustCompile(`^([a-z]+)(?:/(\w+))?$`)
+	reForServiceTargetExpr = regexp.MustCompile(`^([a-z][a-z0-9]*)(?:/(\w+))?$`)
```

Comment:
I'm not a regex expert, but isn't this expression equivalent to `w+`?

Reply:
`\w` would also include uppercase letters.

Reply:
Do we have a specific set of middleware we are trying to capture by using this regex? I am concerned that there might be some other targets which might be an outlier, from the top of my mind I think there is a RDBMS named 4d which will not fit in the above regex.

Assuming we are converting the target type to lowercase somewhere in the pipeline `[a-z0-9]+` might work in most scenarios. WDYT?

Reply:
> Do we have a specific set of middleware we are trying to capture by using this regex?

Got answer to this one from the link in the description, still WDYT about having a more loose regex as `[a-z0-9]+` for capturing service type?

Reply:
+1 on loosening it to `[a-z0-9]+`, that's simpler, I'll update the spec with that.

For reference, here is the quite exhaustive list of databases that are listed in the [OpenTelemetry semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md), while there is none that starts with a number like `4d` there might be in the future.